library/Booster/Pattern/Base.hs

```diff
@@ -113,6 +115,7 @@ data TermAttributes = TermAttributes
     -- variables, recursive through AndTerm
     , hash :: !Int
     , isConstructorLike :: !Bool
+    , canBeConcretelyEvaluated :: Bool
```

Comment:
(nit) why "concretely"? I think `canBeEvaluated` would be OK and shorter. ("evaluated" implying that we use LLVM and that it is concrete). But not too important

Reply:
will change